### PR TITLE
feat: allow detray sterile/non-sterile propagation

### PR DESCRIPTION
**Pull request by asalzburger**
This PR remodels the `detray` Propagation validation, it allows to switch the propagation to `sterile` for timing measurements, and changes the template to stepper while pushing the inspectors and actors to the instantiation of the propagation implantation.

On ODD on can see the effects of the surface grids now:

Without surface grids:
```sh
15:41:18    Sequencer      INFO      Average time per event: 7.804390 ms/event
```

With surface grids:
```sh
15:45:31    Sequencer      INFO      Average time per event: 1.402830 ms/event
```

Great job! @EleniXoch @niermann999

### CODE REVIEW

File: Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp [hunk 1]
```diff
@@ -86,42 +87,76 @@ class DetrayPropagator : public PropagatorInterface {
...
-      Acts::GeometryIdentifier geoID{sf.source()};
-      // Create a step from the object
-      Acts::detail::Step step;
-      step.position = Acts::Vector3(dposition[0], dposition[1], dposition[2]);
-      step.geoID = geoID;
-      step.navDir = object.intersection.direction ? Acts::Direction::Forward
-                                                  : Acts::Direction::Backward;
-      summary.steps.emplace_back(step);
+
+    if (!m_cfg.sterile) {
```
Comment (niermann999) [inline Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp:94]: Maybe this should be split into two classes at some point?

**Review by niermann999 — APPROVED** [review]
Just a few comments that don't need to be addressed in this PR

File: Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp (hunk 1 above)
Comment (asalzburger) [reply Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp:94]: Yes, probably that's better.

File: Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp [hunk 2]
```diff
@@ -35,27 +37,26 @@ using DetrayObjectTracer =
...
-template <typename propagator_t, typename detray_store_t>
+template <typename stepper_t, typename detray_store_t>
 class DetrayPropagator : public PropagatorInterface {
  public:
+  /// Configuration struct
+  struct Config {
+    /// The detray store
+    std::shared_ptr<const detray_store_t> detrayStore = nullptr;
+    /// Switch to sterile
+    bool sterile = false;
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp:48]: I don't use this one yet ... but will include it when needed.

File: Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp [hunk 3]
```diff
@@ -86,42 +87,76 @@ class DetrayPropagator : public PropagatorInterface {
...
+      }
+    } else {
+      // Navigation with inspection
+      using DetrayNavigator =
+          detray::navigator<Acts::DetrayHostDetector,
+                            detray::navigation::default_cache_size>;
+
+      // Propagator with empty actor chain (for the moment)
+      using Propagator =
+          detray::propagator<stepper_t, DetrayNavigator, detray::actor_chain<>>;
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp:144]: Yes, again, I want to have this configurable - as we want to do timing measurements.

File: Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp (hunk 1 above)
Comment (asalzburger) [reply Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp:94]: Let's understand the matrix of propagators we need for full validation and then I will split it into logical pieaces under one `DetrayPropagators.hpp` file which can have the different flavours (we will need material interaction one as well at some point).

**Review by niermann999 — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [6e086127899a40722ff089716a646027fab65e7c]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8216819)!

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047443)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047442)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047441)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047440)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047439)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047438)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047437)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047436)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047435)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047434)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047433)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047432)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047431)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047430)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047429)||
|:red_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047428)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047427)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047426)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047425)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047424)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047423)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047422)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047421)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047420)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047419)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047418)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047417)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047416)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047415)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047414)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047413)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047412)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44047411)||

**Pull request merged**